key/key-test.js

```diff
@@ -1,25 +1,160 @@
 var QUnit = require("steal-qunit");
 var KeyObservable = require("./key");
 var canReflect = require("can-reflect");
+var canReflectDeps = require("can-reflect-dependencies");
+var SimpleMap = require("can-simple-map");
 
 QUnit.module("can-simple-observable/key");
 
 QUnit.test("basics", function(assert) {
-    var value = {foo: {bar: "baz"}};
-    var obs = new KeyObservable(value, "foo.bar");
+	var outer = {inner: {key: "hello"}};
+	var observable = new KeyObservable(outer, "inner.key");
 
-    // Unbound and unobserved behavior
-    assert.equal(canReflect.getValue(obs), "baz", "getValue unbound");
+	// Unbound and unobserved behavior
+	assert.equal(canReflect.getValue(observable), "hello", "getValue unbound");
 
-    canReflect.setValue(obs, "new");
-    assert.equal(canReflect.getValue(value).foo.bar, "new", "value set");
-    assert.equal(canReflect.getValue(obs), "new", "getValue unbound");
+	canReflect.setValue(observable, "aloha");
+	assert.equal(canReflect.getValue(outer).inner.key, "aloha", "value set");
+	assert.equal(canReflect.getValue(observable), "aloha", "getValue unbound");
 });
 
 QUnit.test("get and set Priority", function(assert) {
-    var value = {foo: {bar: "baz"}};
-    var obs = new KeyObservable(value, "foo.bar");
+	var outer = {inner: {key: "hello"}};
+	var observable = new KeyObservable(outer, "inner.key");
+
+	canReflect.setPriority(observable, 5);
+	assert.equal(canReflect.getPriority(observable), 5, "set priority");
+});
+
+QUnit.test("observable has a helpful name", function() {
+	var outer = {inner: {key: "hello"}};
+	var observable = new KeyObservable(outer, "inner.key");
+	QUnit.equal(
+		canReflect.getName(observable),
+		"KeyObservable<Object{}.inner.key>",
+		"observable has the correct name"
+	);
+});
+
+QUnit.test("dependency data", function(assert) {
+	var outer = new SimpleMap({inner: new SimpleMap({key: "hello"})});
+	var observable = new KeyObservable(outer, "inner.key");
+
+	// The observable must be bound before it returns dependency data
+	canReflect.onValue(observable, function() {});
+
+	// Check the observable’s dependency data
+	var observableDepData = canReflectDeps.getDependencyDataOf(observable);
+	assert.notOk(
+		observableDepData.whatChangesMe.mutate,
+		"the observable has no whatChangesMe.mutate dependencies"
+	);
+	assert.equal(
```

Comment:
does `.deepEqual` work here?  I wonder if it would be easier to understand the results if it looked like:

```js
deepEqual(observableDepData, {
  whatChangesMe: {derive: {keyDependencies: new Map([ ... ])}
})
```

Reply:
I'm not sure `map`'s work with `deepEqual` ... but you might be able to then convert it back to the "constructor" equivalent.

Reply:
This works and I think is a little more readable:

```js
	assert.deepEqual(
		observableDepData.whatChangesMe.derive.keyDependencies,
		new Map([
			// the observable is changed by outer’s 'inner' property
			[outer, new Set(["inner"])],
			// the observable is changed by outer.inner’s 'key' property
			[outer.get("inner"), new Set(["key"])]
		]),
		"the observable has the correct whatChangesMe.derive.keyDependencies"
	);
```